### DISCUSSION TITLE
Relation model use function(with) question

### DISCUSSION THREAD

**Discussion by eesaac** [spatie/laravel-permission#2267] [Q&A]
Hi
user relationship queries are not available in the code below.
the problem is config/auth.php in A config function that calls the class as a string.
the model of the guard defined in php is called a string rather than a class.

```php
'providers' => [
        'users' => [
            'driver' => 'eloquent',
            'model' => App\Models\User::class, // <<this call config(auth.providers.{$guard['provider']}.model) is "App\Models\User"
        ],
 ],
```

`Permission::with(['roles', 'users'])->get();`
error:: Class name must be a valid object or a string

```php
public function users(): BelongsToMany
    {
        return $this->morphedByMany(
            \App\Models\User::class, // OK
//            getModelForGuard($this->attributes['guard_name'] ?? config('auth.defaults.guard')), // FAIL '\App\Models\User'
            'model',
            config('permission.table_names.model_has_permissions'),
            PermissionRegistrar::$pivotPermission,
            config('permission.column_names.model_morph_key')
        );
    }
```

how can i this? help!

use package version
"php": "^8.0.2"
"laravel/framework": "^9.19"

**Comment by parallels999** [comment]
https://github.com/spatie/laravel-permission/blob/bc9fff8cb46b13782b07331bf2ab1d13652f1c58/tests/PermissionTest.php#L12-L23
